typescript/radix-sdk/src/modules/tx.ts

```diff
@@ -0,0 +1,519 @@
+import { GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk';
+import {
+  NotarizedTransaction,
+  RadixEngineToolkit,
+  Signature,
+  SignatureWithPublicKey,
+  SimpleTransactionBuilder,
+  TransactionBuilder,
+  TransactionHash,
+  TransactionHeader,
+  TransactionManifest,
+  generateRandomNonce,
+} from '@radixdlt/radix-engine-toolkit';
+
+import { assert } from '@hyperlane-xyz/utils';
+
+import { Account } from '../types.js';
+
+import { RadixPopulate } from './populate.js';
+import { RadixQuery } from './query.js';
+
+export class RadixTx {
+  private account: Account;
+
+  protected query: RadixQuery;
+  protected populate: RadixPopulate;
+  protected networkId: number;
+  protected gateway: GatewayApiClient;
+
+  constructor(
+    account: Account,
+    query: RadixQuery,
+    populate: RadixPopulate,
+    networkId: number,
+    gateway: GatewayApiClient,
+  ) {
+    this.account = account;
+    this.query = query;
+    this.populate = populate;
+    this.networkId = networkId;
+    this.gateway = gateway;
+  }
+
+  public async transfer({
+    to_address,
+    resource_address,
+    amount,
+  }: {
+    to_address: string;
+    resource_address: string;
+    amount: string;
+  }) {
+    const transactionManifest = this.populate.transfer({
+      from_address: this.account.address,
+      to_address,
+      resource_address,
```

Comment:
Is there a way to determine if a resource exists on chain? we could check if it does exists and throw a human friendly error here if it doesn't? or does the transaction fail and the message is clear enough?